### PR TITLE
Ar routing recolculating

### PR DESCRIPTION
**Pull request by BenDenen**

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -169,7 +169,8 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent"
-        tools:background="@color/black">
+        android:visibility="gone"
```
Comment (yunikkk) [inline app/src/main/res/layout/activity_main.xml]: Why is it gone?

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt [hunk 2]
```diff
@@ -48,26 +83,71 @@ class ArNavigationActivity : AppCompatActivity() {
...
+        mapboxNavigation.addOffRouteListener(offRouteListener)
         mapboxNavigation.addProgressChangeListener(mapbox_ar_view)
+        mapboxNavigation.addProgressChangeListener(progressChangeListener)
+        mapboxNavigation.locationEngine = arLocationEngine
         mapboxNavigation.startNavigation(intent.getSerializableExtra(EXTRA_ROUTE) as DirectionsRoute)
     }
 
     override fun onPause() {
-        super.onPause()
+        arLocationEngine.apply{
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: Format needed

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt [hunk 3]
```diff
@@ -48,26 +83,71 @@ class ArNavigationActivity : AppCompatActivity() {
         back.setOnClickListener {
             onBackPressed()
         }
+
+        val builder = MapboxNavigationOptions
+                .builder()
+                .enableOffRouteDetection(true)
+        mapboxNavigation = MapboxNavigation(this, Mapbox.getAccessToken()!!, builder.build())
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: Can store accessToken to variable.

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt [hunk 4]
```diff
@@ -29,16 +45,35 @@ class ArNavigationActivity : AppCompatActivity() {
         }
     }
 
-    private val mapboxNavigation by lazy {
-        MapboxNavigation(this, Mapbox.getAccessToken()!!)
+    private val arLocationEngine by lazy {
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: Why do we need `locationEngine` here?

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (BenDenen) [reply app/src/main/res/layout/activity_main.xml]: Initially we should not show it

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt (hunk 4 above)
Comment (BenDenen) [reply app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: We need it to make Navigation work properly. In opposite way I will create default location provider which does not work well and we will miss offRoad event

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt (hunk 3 above)
Comment (BenDenen) [reply app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: Yes we can but Mapbox should be initialised at that moment so I think it is one more checking for it

Comment (yunikkk) [reply app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: I mean we can `val token = Mapbox.getAccessToken()!!` and then reuse the variable here and below

**Pull request merged**